genai/README.md

```diff
@@ -1,189 +1,159 @@
-# LangChain AI Service with Hono
+# Prompteus GenAI Service
+
+AI-powered service for analyzing GitHub contributions and providing intelligent Q&A about developer weekly summaries.
+
+## 🚀 What Works
+
+### Core Functionality
+- **Contribution Ingestion**: Store GitHub contributions with user/week organization
+- **AI Summary Generation**: LangChain/LangGraph-powered structured summaries with streaming support
+- **Context-Aware Q&A**: Ask questions about contributions with evidence and confidence scoring
+- **Semantic Search**: Meilisearch integration for finding relevant contributions
+- **Prometheus Metrics**: Comprehensive observability and monitoring
+- **Task-based Processing**: Async ingestion with progress tracking
+
+### API Endpoints
+For complete API documentation, see the **[Scalar API Reference](/reference)** when the service is running.
+
+## 🏗️ How It Works
+
+### Service Architecture
+
+```mermaid
+graph TB
+    subgraph "API Layer"
+        API[FastAPI App]
+        DEPS[Dependency Injection]
+    end
+    
+    subgraph "Business Services"
+        ING[ContributionsIngestionService]
+        SUM[SummaryService]
+        QA[QuestionAnsweringService]
+    end
+    
+    subgraph "Data & AI"
+        MEILI[Meilisearch Service]
+        LC[LangChain/LangGraph]
+        LLM[OpenAI GPT]
+    end
+    
+    subgraph "Observability"
+        METRICS[Prometheus Metrics]
+        LOGS[Structured Logging]
+    end
+    
+    API --> DEPS
+    DEPS --> ING
+    DEPS --> SUM
+    DEPS --> QA
+    
+    ING --> MEILI
+    SUM --> ING
+    SUM --> LC
+    QA --> ING
+    QA --> LC
+    QA --> MEILI
+    
+    LC --> LLM
+    
+    API --> METRICS
+    API --> LOGS
+    
+    style API fill:#e1f5fe
+    style LC fill:#f3e5f5
+    style MEILI fill:#fff3e0
+    style METRICS fill:#ffebee
+```
 
-This project provides a containerized LangChain-based AI service built with Hono (NodeJS), orchestrated via Docker Compose. It features live reload for development and an optimized production image.
+### Data Flow
+
+```mermaid
```

Comment:
We should in todays meeting discuss whcih API endpoints we need. I think we don't quit sync on this 😉
Master got its own brand of different APIs, but I like yours better.